blocklib/picoscope5000a/picoscope5000a/picoscope5000a.yml

```diff
@@ -0,0 +1,132 @@
+module: picoscope5000a
+block: picoscope5000a
+label: PicoScope 5000a
+blocktype: sync_block
+includes:
+  - gnuradio/digitizers/digitizer_enums.h
+  - gnuradio/picoscope5000a/enums.h
+
+parameters:
+-   id: serial_number
+    label: Serial Number
+    dtype: string
+-   id: sample_rate
+    label: Sample Rate (Hz)
+    dtype: rf64
+    default: 10000
+-   id: buffer_size
+    label: Buffer Size
+    dtype: size
+    default: 8192
+-   id: nr_buffers
+    label: Number of Buffers
+    dtype: size
+    default: 100
+-   id: driver_buffer_size
+    label: Driver Buffer Size
+    dtype: size
+    default: 100000
+-   id: pre_samples
+    label: Pre-Sample Count
+    dtype: size
+    default: 1000
+-   id: post_samples
+    label: Post-Sample Count
+    dtype: size
+    default: 9000
+-   id: acquisition_mode
+    label: Acquisition Mode
+    dtype: enums/digitizer_acquisition_mode_t
+    is_enum: true
+    default: digitizer_acquisition_mode_t::STREAMING
+-   id: rapid_block_nr_captures
+    label: No. of Captures (Rapid Block Mode)
+    dtype: size
+    default: 1
+-   id: streaming_mode_poll_rate
+    label: Polling rate (Streaming Mode)
+    dtype: rf64
+    default: 0.001
+-   id: downsampling_mode
+    label: Downsampling Mode
+    dtype: enums/digitizer_downsampling_mode_t
+    is_enum: true
+    default: digitizer_downsampling_mode_t::NONE
+-   id: downsampling_factor
+    label: Downsampling Factor
+    dtype: size
+    default: 1
+-   id: auto_arm
+    label: Auto-Arm
+    dtype: bool
+    default: 'true'
+-   id: trigger_once
+    label: Trigger once
+    dtype: bool
+    default: 'false'
+# PS5000A-specific
+-   id: resolution
+    label: Device Resolution
+    dtype: enums/digitizer_resolution_t
+    is_enum: true
+    default: digitizer_resolution_t::R_8BIT
+
+callbacks:
+   - id: hardware_version
+     const: true
+     return: string
+   - id: driver_version
+     const: true
+     return: string
+   - id: actual_sample_rate
+     const: true
+     return: rf64
+   - id: initialize
+     return: void
+   - id: close
+     return: void
+# TODO(PORT) maybe makes these parameters (list of structs)
```

Comment:
If that is possible, it would be nice !   Though I suppose it's like this as well for the other digitizer,  so imo rather a refactoring which should be done afterward, in a separate MR